dtk/process.py

```diff
@@ -525,9 +532,82 @@ def nextpow2(i):
     except IndexError:
         amplitude = 2*abs(Y[1:n//2])
         # power = abs(Y[1:n/2])**2
+
     return frequency, amplitude
 
 
+def pow_spectrum(data, sampleRate):
+    """
+    Return the power spectrum of a dataset.
+    
+    S(f) = |X(f)|^2
+
+    Parameters
+    ----------
+    data : ndarray, shape (m,) or shape(n,m)
+        The array of time signals where n is the number of variables and m is
+        the number of time steps.
+    sampleRate : int
+        The signal sampling rate in hertz.
+    
+
+    Returns
+    -------
+    frequency : ndarray, shape (p,)
+        The frequencies where p is a power of 2 close to m.
+    power : ndarray, shape (p,n)
+        The power at each frequency.
+
```

Comment:
If you can, it would be nice to add these plot directives to show an simple example.

Reply:
Done! Please check if it works as expected. I am not experienced with using plot directives.